redis/tests/support/sentinel.rs

```diff
@@ -0,0 +1,271 @@
+    )
+}
+
+fn spawn_replica_server(
+    port: u16,
+    master_port: u16,
+    dir: &TempDir,
+    modules: &[Module],
+) -> RedisServer {
+    let config_file_path = dir.path().join("redis_config.conf");
+    File::create(&config_file_path).unwrap();
+
+    RedisServer::new_with_addr(
+        redis::ConnectionAddr::Tcp("127.0.0.1".into(), port),
+        Some(&config_file_path),
+        None,
+        modules,
+        |cmd| {
+            cmd.arg("--replicaof")
+                .arg("127.0.0.1")
+                .arg(master_port.to_string());
+            // Minimize startup delay
+            // cmd.arg("--repl-diskless-sync-delay").arg("0");
+            // cmd.arg("--repl-ping-replica-period").arg("1");
+            cmd.arg("--appendonly").arg("yes");
+            // cmd.stdout(std::process::Stdio::inherit())
+            //     .stderr(std::process::Stdio::inherit());
+            // cmd.arg("--loglevel").arg("debug");
+            cmd.current_dir(dir.path());
+            cmd.spawn().unwrap()
+        },
+    )
+}
+
+fn spawn_sentinel_server(
+    port: u16,
+    master_ports: &[u16],
+    dir: &TempDir,
+    modules: &[Module],
+) -> RedisServer {
+    let config_file_path = dir.path().join("redis_config.conf");
+    let mut file = File::create(&config_file_path).unwrap();
+    for (i, master_port) in master_ports.iter().enumerate() {
+        file.write_all(
+            format!("sentinel monitor master{} 127.0.0.1 {} 1\n", i, master_port).as_bytes(),
+        )
+        .unwrap();
+        // file.write_all(format!("sentinel down-after-milliseconds master{} 200\n", i).as_bytes())
+        //     .unwrap();
+        // file.write_all(format!("sentinel parallel-syncs master{} 1\n", i).as_bytes())
+        //     .unwrap();
+        // file.write_all(format!("sentinel failover-timeout master{} 5000\n", i).as_bytes())
+        //     .unwrap();
+    }
+    file.flush().unwrap();
+
+    RedisServer::new_with_addr(
+        redis::ConnectionAddr::Tcp("127.0.0.1".into(), port),
+        Some(&config_file_path),
+        None,
+        modules,
+        |cmd| {
+            // cmd.stdout(std::process::Stdio::inherit())
+            //     .stderr(std::process::Stdio::inherit());
+            // cmd.arg("--loglevel").arg("debug");
+            cmd.arg("--sentinel");
+            cmd.arg("--appendonly").arg("yes");
+            cmd.current_dir(dir.path());
+            cmd.spawn().unwrap()
+        },
+    )
+}
+
+impl RedisSentinelCluster {
+    pub fn new(masters: u16, replicas_per_master: u16, sentinels: u16) -> RedisSentinelCluster {
+        RedisSentinelCluster::with_modules(masters, replicas_per_master, sentinels, &[])
+    }
+
+    pub fn with_modules(
+        masters: u16,
+        replicas_per_master: u16,
+        sentinels: u16,
+        modules: &[Module],
+    ) -> RedisSentinelCluster {
+        let mut servers = vec![];
+        let mut folders = vec![];
+        let mut master_ports = vec![];
+        let start_port = 7000;
```

Comment:
don't use a hard-coded port. Find an available port, like [here](https://github.com/redis-rs/redis-rs/blob/6be07049f4982af3b34c1be5cbd7cc0b7a06c6c0/redis/tests/support/mod.rs#L100).

Better that you find an available port for each node, too, instead of assuming that you have enough available ports in a row.

Reply:
I agree, I just did it like this because the cluster tests are like this, and I copied some code from it: https://github.com/redis-rs/redis-rs/blob/6be07049f4982af3b34c1be5cbd7cc0b7a06c6c0/redis/tests/support/cluster.rs#L78

I'll try and change to use a random port now.

Reply:
It was far easier than I initially thought, done!